httpcore5/src/main/java/org/apache/hc/core5/http/ContentType.java

```diff
@@ -288,8 +333,19 @@ public String toString() {
         buf.append(this.mimeType);
         if (this.params != null) {
             buf.append("; ");
-            MessageSupport.formatParameters(buf, this.params);
-        } else if (this.charset != null) {
+            boolean first = true;
+            for (final NameValuePair param : params) {
```

Comment:
@arturobernalg One could use normal index loop here and avoid this super ugly `first` variable (and likely be more memory efficient).

Reply:
HI @ok2c 
The ´first´ variable is needed to handle the condition correctly, ensuring that parameters are properly separated while avoiding an extra "; " in cases where the charset is implicit.

Reply:
> HI @ok2c The ´first´ variable is needed to handle the condition correctly, ensuring that parameters are properly separated while avoiding an extra "; " in cases where the charset is implicit.

@arturobernalg You are right. Indeed, an index variable would not work here. Still, I would avoid using `for each` loops in HC.

Reply:
How about the join() API?

Reply:
@ok2c  fair enough. changed